### ISSUE TITLE
Comprehensive sound ideas & bug fixes topic

### ISSUE THREAD

**Issue by 787992** [GaijinEntertainment/fmod_studio_warthunder_for_modders#142]
Hello! I'm not really sure what future plans you have guys, but let me stop by around some ideas gathered here for several years.

1. [Corrected engine events for tanks](https://github.com/GaijinEntertainment/fmod_studio_warthunder_for_modders/issues/128) - based on .blk engine names, fixed most misused tank engines and corrected forgotten ones based on our *list*, you can ignore low tier tanks, that was based on rational fidelity

2. [Corrected ATGM list](https://github.com/GaijinEntertainment/fmod_studio_warthunder_for_modders/issues/130)- also based on *list* this is only subjective suggestion most of misused stuff is fixed, but still here for the sake of divercity

3. [Rapid explosiove sounds](https://github.com/GaijinEntertainment/fmod_studio_warthunder_for_modders/issues/122) for guns with rpm above 1500-2000 w/caliber 20mm and above - pretty much selfexplanatory, I think this is missed opportunity to make more depth and sound reward for brrt guns or close to it rpm wise

4. Helicopters desperately *needs* ai sounds per engine - code land is ready for implementing these sounds since 2019(?), Helicopters are big thread on the battlefield and knowing what type of heli flying to rush tanks is useful and must have feature.

5. All aviation really needs new Rocket launch and ATGM launch sounds, right now this is extremely simple sound that plays in exterior and doesn't even have interior version. This is undewhelming at most.

6. Reload sounds diversity - we have some new reload sound for top tier tanks, but never had anything for small and high caliber cannons, sound event for howitzers even existing in sound list but never used in the game, which is actually plays big role for immersion and depth, but I don't think all tanks should have one exact same reload sound, so just split reloads from 37mm to 57mm as small, 75mm to 120mm as mid and 130mm to 152mm as big.

7. Parameter for shell extracting - big amount of tanks using auto-loader while using sharing interior sounds with other tanks, this simple idea is just prevent playing shell extraction layer while shooting from cannons with auto-loader
![image](https://user-images.githubusercontent.com/63158562/188490807-70eb160a-1862-4cd8-af09-cf0c8862823b.png)

8. Parameter for open-top tanks or glass cannons - exact same as previous one but with exception of using some internal layers, obviously tanks like Chi-ha LG/Ho-Ro or Flak w/kwk43 can't sound like shoot was made from enclosed turret.
![image](https://user-images.githubusercontent.com/63158562/188490958-c938e446-3a8b-48ed-9619-4717cb42b421.png)

9. Prevent hearing interior noises right after switching from sight to exterior view, this is very old suggestion: people almost never hearing hard work of exterior cannon shots because after switching from sight to exterior you can hear only exterior noises.

Made little video explaing the main idea

https://user-images.githubusercontent.com/63158562/188491899-220adf5c-9359-4666-b872-7da62e190162.mp4

Currently it's not possible due to how sound spawn system works: when you start moving after shot sound source goes away

https://user-images.githubusercontent.com/63158562/188491922-4ce9b15b-89dd-46d7-859d-dcd8f27b4adc.mp4

Bugs:

- Please take a look at armor hits sounds, mid-low calibers always playing non-pen sound events (ai events)
- When artillery hits any metal surface - armor penetration sound plays (instead of intened arty sound event for metal)
- Not sure if it was fixed or not, GAU8 (ai) cannon ignores speed of sound feature

**Comment by Uchuujin01** [comment]
Hello! Uh, quite a lot ) Thank you for gather here all your suggestions. Sorry for I still did nothing on them although you suggested some of them long time ago. I like them mostly, but needed to postpone them since we had a lot of tasks on sound.

I know I wrote this before for a several times, but, I'll return to those topics after the upcoming major patch ) Thank you again for your suggestions, they really about to improve game sound, I completely understand this.

**Comment by OXEPU3M** [comment]
> Currently it's not possible due to how sound spawn system works: when you start moving after shot sound source goes away

Removing 3D timeline with 2D doesn't help, right?
Just curious.

**Comment by 787992** [comment]
> > Currently it's not possible due to how sound spawn system works: when you start moving after shot sound source goes away
> 
> Removing 3D timeline with 2D doesn't help, right? Just curious.

No, to use distance parameter properly, you need to set this as 3D

**Comment by OXEPU3M** [comment]
> No, to use distance parameter properly, you need to set this as 3D

Why does it need Distance if you shot with interior view? Also when you are shooting 3d person it doesn't matter too. Definitely only **_AI**'s need this parameter.

**Comment by 787992** [comment]
> > No, to use distance parameter properly, you need to set this as 3D
> 
> Why does it need Distance if you shot with interior view? Also when you are shooting 3d person it doesn't matter too. Definitely only **_AI**'s need this parameter.

Ah you mean 3D as directional, no this is not directional. The sound playing everywhere (mix 100%) but using distance parameter with spatialilizer.
The main problem is that source of sound just not attached to your tank, and when you moving it will follow with slight delay, hence it works perfect when standing still

**Comment by 787992** [comment]
@Uchuujin01 what's speed_relative parameter actually does outside of simulating sonic boom for planes? It feels like it counting difference of speed between player and ai, but it works from 0 to 100 despite the timeline up to 800, it's really useful parameter though
